cmd/outline-ss-server/main.go

```diff
@@ -346,8 +346,9 @@ func (s *OutlineServer) runConfig(config Config) (func() error, error) {
 								slog.Error("failed to upgrade", "err", err)
 							}
 							defer conn.Close()
-							if clientIP := net.ParseIP(r.RemoteAddr); clientIP != nil {
-								conn = &replaceAddrConn{StreamConn: conn, raddr: &net.TCPAddr{IP: clientIP}}
+							clientAddrPort, err := onet.ParseAddrPortOrIP(r.RemoteAddr)
+							if err == nil {
+								conn = &replaceAddrConn{StreamConn: conn, raddr: net.TCPAddrFromAddrPort(clientAddrPort)}
```

Comment:
Let's add a TODO to replace this with a client request. Then we can make it have the client IP only, no need for port and wrapping.